src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs

```diff
@@ -4,31 +4,14 @@
 
 // Zip Spec here: http://www.pkware.com/documents/casestudies/APPNOTE.TXT
 
-using System.Collections.Generic;
 using System.Collections.ObjectModel;
-using System.Diagnostics;
 using System.Text;
 
 namespace System.IO.Compression
 {
     public class ZipArchive : IDisposable
     {
-        private readonly Stream _archiveStream;
-        private ZipArchiveEntry? _archiveStreamOwner;
-        private BinaryReader? _archiveReader;
-        private ZipArchiveMode _mode;
-        private List<ZipArchiveEntry> _entries;
-        private ReadOnlyCollection<ZipArchiveEntry> _entriesCollection;
-        private Dictionary<string, ZipArchiveEntry> _entriesDictionary;
-        private bool _readEntries;
-        private bool _leaveOpen;
-        private long _centralDirectoryStart; //only valid after ReadCentralDirectory
-        private bool _isDisposed;
-        private uint _numberOfThisDisk; //only valid after ReadCentralDirectory
-        private long _expectedNumberOfEntries;
-        private Stream? _backingStream;
-        private byte[]? _archiveComment;
-        private Encoding? _entryNameEncoding;
+        internal IZipArchiveStrategy Strategy { get; }
```

Comment:
This is a new allocation.

Reply:
It's similar to what we have in FileStream. If we find places where we can improve perf, it can compensate this extra allocation. Unfortunately, due to the inheritance of the abstract class, I cannot switch to using structs. Do you have any alternatives?